doc/tutorial/c_tutorial/c_tutorial.rst

```diff
@@ -0,0 +1,148 @@
+********************************
+WIP: How to befriend DDlog and C
+********************************
+
+.. contents::
+
+Prerequisites
+=============
+
+First, you need to obtain DDlog__ from the official repository, which contains instructions how to install and set it up. There is also an expectation that you are familiar with DDlog syntax and commands, i.e., read the tutorial or at least skim through it.
+
+__
+.. _DDlog DDlog repository on GitHub:
+    https://github.com/vmware/differential-datalog
+
+Introduction
+============
+
+In this set of tutorials we are going to explore various features of DDlog and learn how to befriend DDlog program with C program. To do that we are going to build several simple yet functional applications as it is more fun to build real programs. Furthermore, it will also demonstrate that DDlog is not limited to a particular domain, but rather can be used to solve different problems.
+
+Tutorial #1 - Reachibility Monitor
+==================================
+
+In our first tutorial we are going to build a program that will store the reachibility information in the network of four nodes: Menlo Park, Santa Barbara, Los Angeles, and Salt Lake City. User will update link status between nodes through CLI and our program will calculate the rest. The purpose of this program is to provide a simple interface to a network operator and return information regarding which node can reach which. Thus, the only concern of network operator is to update link status without having to worry about doing those calculations themself. While, this task can be relatively easy done in our example of four nodes, in a more realistic scenario, the number of nodes can be in the order of hundreds and link status updates can be streamed into the monitoring application.
+
+Let's start with defining our program in the DDlog language. In our program we need to supply source node, destination node, and a link status between them. Now we are ready to define our input relation :code:`Links`. Let's create a file called :code:`t1_reachibility_monitor.dl` and add there a following line representing input relation for our reachibility monitor:
+
+.. code-block::
+    
+	input relation Links(src: string, dst: string, link_status: bool)
+
+However, for the reachibility monitor to be useful it also should provide some output and not just ingest the data. We want our program to output a list of nodes that are currently reachable based on the information in the relation :code:`Links`. Thus, let's add an output relation :code:`ConnectedNodes` and provide a rule to calculate records in this relation. To do that we are going to add few more lines to the :code:`t1_reachibility_monitor.dl` so it will looks as follows:
+
+.. code-block::
+
+	input relation Links(src: string, dst: string, link_status: bool)

```

Comment:
I think you are using tabs. does .rst do something special to tabs? Otherwise I would suggest using spaces to get a predictable layout.